NYCPlanning/data-engineering
# start testing KPDB builds

resolves #518, related to #547

This adds data tests for the primary KPDB table using `dbt`. It also starts to test some of the aggregation tables. These could be considered loose data contracts ([docs](https://docs.getdbt.com/docs/collaborate/govern/model-contracts)).

Many of the filename and directory changes are because `dbt` expects the configuration yml files to be in subdirectories ([docs](https://docs.getdbt.com/reference/model-properties)) and expects to find sql filenames matching the names of tables defined in config files.

I'll follow up on this with more tests to investigate and fix issues with KPDB data found by GIS.

## changes
- use latest KPDB corrections
- ensure aggregations use final KPDB table
- move and consolidate aggregation-related scripts
- create primary KPDB table before aggregations
- add and use dbt to test high priority KPDB tables

## results of changes

builds using this branch [here](https://github.com/NYCPlanning/data-engineering/actions/workflows/build.yml?query=branch%3Adm-kpdb-for-gis)

latest example of testing primary KPDB table [here](https://github.com/NYCPlanning/data-engineering/actions/runs/7744371614/job/21117967250#step:10:61) and aggregation tables [here](https://github.com/NYCPlanning/data-engineering/actions/runs/7745995963/job/21123203139#step:12:58)
